package/kedro_viz/models/flowchart/entities.py

```diff
@@ -0,0 +1,30 @@
+from pydantic import BaseModel, Field, ValidationInfo, field_validator
+from typing import Optional
+
+
+class NamedEntity(BaseModel):
+    """Represent a named entity (Tag/Registered Pipeline) in a Kedro project."""
+
+    id: str
+    name: Optional[str] = Field(
+        default=None,
+        validate_default=True,
+        description="The name of the entity",
+    )
+
+    @field_validator("name")
+    @classmethod
+    def set_name(cls, _, info: ValidationInfo):
+        assert "id" in info.data
+        return info.data["id"]
+
+
+class RegisteredPipeline(NamedEntity):
```

Comment:
Hi @SajidAlamQB , most of the splitting looks good. Some suggestions -

On first look, entities file is not obvious on what it contains but the class names make sense here. I would have the below structure -

- flowchart 
    - nodes
    - node_metadata 
    - edge
    - pipelines
    - tag
    - model_utils 

* NamedEntity can be in entity or model_utils
* RegisteredPipeline can be in pipelines and Tag in its own tag file. 
* Model file names are singular if we are not having multiple classes (nodes and pipelines can be plural as they have collection of classes)

Thank you

Reply:
`GraphNodeTypes` and `ModularPipelineChild` are needed by `nodes.py` and `pipelines.py` to avoid circular imports I have moved them into `model_utils`.

Reply:
ModularPipelineNode is present in both nodes.py and pipelines.py. I am not sure why, maybe this is causing the circular imports ?

Reply:
Yes, it was thanks!